test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+               components: components,
+               security: security,
+               tags: tags,
+               externalDocs: externalDocs,
+               extensions: extensions
+             } = spec
+
+      assert "3.0.0" == openapi
+
+      # OpenApi Info Object
+      assert %OpenApiSpex.Info{
+               title: _title,
+               version: _version,
+               description: _description,
+               contact: contact,
+               license: license
+             } = info
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.Contact{} = contact
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.License{} = license
+
+      # OpenApi Specification Extensions
+      assert %{
+               test: true
+             } == extensions
+
+      # OpenApi External Documentation Object
+      assert %OpenApiSpex.ExternalDocumentation{
+               description: _,
+               url: _
+             } = externalDocs
+
+      # OpenApi Components Object
+      assert %OpenApiSpex.Components{
+               callbacks: callbacks,
+               schemas: schemas,
+               responses: responses,
+               examples: examples,
+               links: %{
+                 "address" => link
+               },
+               requestBodies: requestBodies,
+               parameters: %{
+                 "AcceptEncodingHeader" => components_parameters_parameter
+               },
+               securitySchemes: securitySchemes,
+               headers: %{
+                 "api-version" => components_headers_header
+               }
+             } = components
+
+      assert %{
+               "test" => %OpenApiSpex.RequestBody{
+                 description: "user to add to the system",
+                 content: %{
+                   "application/json": media_type
+                 },
+                 required: false
+               }
+             } = requestBodies
+
+      assert %OpenApiSpex.MediaType{
+               schema: %OpenApiSpex.Reference{},
+               examples: %{
+                 user: %OpenApiSpex.Example{}
+               },
+               encoding: %{
+                 historyMetadata: %OpenApiSpex.Encoding{
+                   contentType: "application/xml; charset=utf-8",
+                   style: :simple,
+                   explode: false,
+                   allowReserved: false,
+                   headers: %{
+                     "X-Rate-Limit-Limit": %OpenApiSpex.Header{
+                       description: "The number of allowed requests in the current period",
+                       schema: %OpenApiSpex.Schema{
+                         type: :integer
+                       }
+                     }
+                   }
+                 }
+               },
+               example: nil
+             } = media_type
+
+      assert %{
+               "componentCallback" => componentCallback
+             } = callbacks
+
+      assert %{
```

Comment:
`Callback` should be a map with string keys: https://github.com/GhostGroup/open_api_spex/blob/2cbf78b61693808b803292d93c64a435fd860bae/lib/open_api_spex/callback.ex#L17